src/rgen/Microsoft.Macios.Generator/ExtraSources.cs

```diff
@@ -0,0 +1,35 @@
+namespace Microsoft.Macios.Generator;
+
+public static class ExtraSources {
+
+	/// <summary>
+	/// Attribute to be added at init time that will allow to mark classes as binding types.
+	/// </summary>
+	const string bindingTypeAttributeSourceCode = @"// <auto-generated/>
+using System;
+
+namespace Foundation 
+{
+    [System.AttributeUsage(System.AttributeTargets.Class | System.AttributeTargets.Enum, AllowMultiple = false)]
+    public class BindingTypeAttribute : System.Attribute
+    {
+        public string Name { get; set; }
+        public Type [] Events { get; set; }
+        public string [] Delegates { get; set; }
+        public bool Singleton { get; set; }
+
+        // If set, the code will keep a reference in the EnsureXXX method for
+        // delegates and will clear the reference to the object in the method
+        // referenced by KeepUntilRef.   Currently uses an ArrayList, so this
+        // is not really designed as a workaround for systems that create
+        // too many objects, but two cases in particular that users keep
+        // trampling on: UIAlertView and UIActionSheet
+        public string KeepRefUntil { get; set; }
```

Comment:
Injecting it allows the users not to reference any of the dll, so they can do findings without our dll. I'll change the namespace, but we might want to use a more language agnostic namespace since we can use the generator to write swift bindings too.

Reply:
> Injecting it allows the users not to reference any of the dll, so they can do bindings without our dll.

That doesn't make much sense: where do you get `NSObject` from for instance?

Also it won't work when bindings reference our dll, because then the type will be in two places.

Reply:
> we can use the generator to write swift bindings too.

Swift bindings are going to be done very differently, so I'd focus on Objective-C for this.